controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1738,6 +1743,18 @@ func saveHostProvisioningSettings(host *metal3api.BareMetalHost, info *reconcile
 	return
 }
 
+func saveHostFirmwareComponents(hfc *metal3api.HostFirmwareComponents, info *reconcileInfo) error {
+
+	if !reflect.DeepEqual(hfc.Status.Updates, hfc.Spec.Updates) {
+		info.log.Info("Updates for HostFirmwareComponents have changed")
+		hfc.Status.Updates = hfc.Spec.Updates
+		t := metav1.Now()
+		hfc.Status.LastUpdated = &t
+		return nil
```

Comment:
Without calling `Update()` this doesn't have any effect.

Reply:
You are talking about having the reconciler calling Status().Update() right ?  `r.Status().Update(info.ctx, info.hfc)`
Any ideas on how to access the HostFirmwareComponentsReconciler? Or there is other way to call

Reply:
Since this function is in baremetalhost controller, you can either define it as a method to BMH Reconciler or call the `Status().Update` in the caller function.
By the way, in line 1750 are we sure that all the updates have been applied by this point ?

Reply:
@hroyrh ohhh. that makes some sense! 
Regarding L1750,  since this is in the `saveHostFirmwareComponents` I should probably re-think the place I'm calling it actionPreparing (L1174 atm), maybe I should move it to L118.. so we would be sure that there is nothing else to do and we would do actionComplete right after it..